java/code/src/com/suse/manager/webui/services/impl/SystemQuery.java

```diff
@@ -0,0 +1,498 @@
+    Optional<List<MgrK8sRunner.Container>> getAllContainers(String kubeconfig, String context);
+
+    /**
+     * Remove SUSE Manager specific configuration from a Salt minion.
+     *
+     * @param minion the minion.
+     * @param timeout operation timeout
+     * @return list of error messages or empty if no error
+     */
+    Optional<List<String>> cleanupMinion(MinionServer minion, int timeout);
+
+    /**
+     * Send notification about a system id to be generated.
+     * @param minion target minion.
+     * @throws InstantiationException if signature generation fails
+     * @throws SaltException if anything goes wrong.
+     */
+    void notifySystemIdGenerated(MinionServer minion) throws InstantiationException, SaltException;
+
+    /**
+     * Get pending resume information.
+     * @param minionIds to target.
+     * @return pending resume information.
+     * @throws SaltException if anything goes wrong.
+     */
+    Map<String, Result<Map<String, String>>> getPendingResume(List<String> minionIds) throws SaltException;
+
+    /**
+     * Return show highstate result.
+     * @param minionId of the target minion.
+     * @return show highstate result.
+     * @throws SaltException if anything goes wrong.
+     */
+    Map<String, Result<Object>> showHighstate(String minionId) throws SaltException;
+
+    /**
+     * Query product information.
+     * @param minionId of the target minion.
+     * @return product information
+     */
+    Optional<List<Zypper.ProductInfo>> getProducts(String minionId);
+
+    /**
+     * Execute a LocalCall asynchronously on the default Salt client.
+     *
+     * @deprecated this function is too general and should be replaced by more specific functionality.
+     * @param <T> the return type of the call
+     * @param callIn the call to execute
+     * @param target minions targeted by the call
+     * @param metadataIn the metadata to be passed in the call
+     * @return the LocalAsyncResult of the call
+     * @throws SaltException in case of an error executing the job with Salt
+     */
+    @Deprecated
+    <T> Optional<LocalAsyncResult<T>> callAsync(LocalCall<T> callIn, Target<?> target,
+                                                       Optional<ScheduleMetadata> metadataIn) throws SaltException;
+
+    /**
+     * Sync the channels of a list of minions
+     * @param minionIds of the targets.
+     * @throws SaltException if anything goes wrong.
+     */
+    void deployChannels(List<String> minionIds) throws SaltException;
+
+    /**
+     * Upload built Kiwi image to SUSE Manager
+     *
+     * @param minion     the minion
+     * @param filepath   the filepath of the image to upload, in the build host
+     * @param imageStore the image store location
+     * @return the execution result
+     */
+    Optional<MgrUtilRunner.ExecResult> collectKiwiImage(MinionServer minion, String filepath,
+                                                               String imageStore);
```

Comment:
This is really not important, but I would prefer if we would use a unified way of doing indentation. In this new interface we already use 3 different ways of indenting the arguments: 8 spaces after the method name (my least preferred one), 8 spaces starting from the beginning of the line and 4 spaces from the beginning of the line one argument per line. It would be a good opportunity now to use just one of those ways, at least throughout this new interface.